extensions/wikia/AjaxPoll/AjaxPoll_body.php

```diff
@@ -149,54 +147,48 @@ public function getVotes($isSubmit = false) {
 		$votes = array();
 		$total = 0;
 
-		$memcKey = wfMemcKey(self::MEMC_PREFIX_GETVOTES, $this->mId);
-		if (!$isSubmit) {
-			$votes = $wgMemc->get($memcKey);
+		$memcKey = $this->getVotesMemKey();
+		if ( !$isSubmit ) {
+			$votes = $wgMemc->get( $memcKey );
 		}
-		if (!empty($votes)) {
-			foreach($votes as $ansData){
+		if ( !empty( $votes ) ) {
+			foreach( $votes as $ansData ){
 				$total += $ansData["value"];
 			}
 		} else {
-			$votes = array();
 			$dbr = wfGetDB( $isSubmit ? DB_MASTER : DB_SLAVE );
```

Comment:
The primary improvement of this change is to do a ```group by``` in SQL rather than pulling all rows back and doing a "code" group by in PHP.